### DISCUSSION TITLE
Using Wallet Connect as part of Solana Wallet Adapter issues

### DISCUSSION THREAD

**Discussion by Pure-lmao** [WalletConnect/walletconnect-monorepo#2481] [Developer Support]
Hey

We currently have the Solana Wallet Adapter in use on our site but recently became aware that the Wallet Connect option was not working: the QR code would not pop up. 
We realised that there was additional setup needed to have Wallet Connect work, while the other wallets just work without any additional work.

Following the example here: https://www.npmjs.com/package/@solana/wallet-adapter-walletconnect, I added those params to our `new WalletConnectWalletAdapter()` instance. The projectId is taken from the WC Cloud site and the project is currently a draft. 

Now the error in the console is `Error: WebSocket connection closed abnormally with code: 3000 (Authorization error: Project ID is missing
`. I have spent a long time looking through the Solana Wallet Adapter and WC docs, but there seems to be no additional documentation on how this work. 

It has already been suggested to use Solib or universal-provider, but since the SWA works well enough and has been in use for a long time, I would rather fix the issue with it. I have also looked through forums and github discussions and I see no one else mentioning this or having this error so I am quite confused.

**Comment by Pure-lmao** [comment]
I have messed about more and got to the point where that error sometimes does not occur, but instead I get `Error: socket stalled` followed by `WalletConnectionError: closeTransport called before connection was established`. The QR code does not appear.

**Reply by finessevanes** [reply]
What SDK and version are you using?

**Comment by johnnydepp9** [comment]
I have same issue, @Pure-lmao  Do you fixed?

**Reply by Pure-lmao** [reply]
No, I gave up and removed the WC option from the SWA. Then they closed the WC cloud account or something because we didn't have WC v2 on the website. Some day, we'll have a dev who knows the esoteric ways of WC and they can fix it.
